src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/UserManagement/UserManagementController.cs

```diff
@@ -0,0 +1,142 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Admin.ViewModels;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Controllers.Api.UserManagement
+{
+    [ApiController]
+    [Route("api/user-management")]
+    [Authorize(Policy = "CanManageUsersOnSystem")]
+    public class UserManagementController : ControllerBase
+    {
+        private readonly IUserManagementService _userManagementService;
+        private readonly IUserRoleService _userRoleService;
+
+        public UserManagementController(IUserManagementService userManagementService,
+            IUserRoleService userRoleService)
+        {
+            _userManagementService = userManagementService;
+            _userRoleService = userRoleService;
+        }
+
+        [HttpGet("users")]
+        [ProducesResponseType(200)]
+        public async Task<ActionResult<List<UserViewModel>>> GetUserList()
+        {
+            return await _userManagementService
+                .ListAllUsers()
+                .HandleFailuresOrOk();
+        }
+
+        [HttpGet("users/{id:guid}")]
+        [ProducesResponseType(200)]
+        public async Task<ActionResult<UserViewModel>> GetUser(Guid id)
+        {
+            return await _userManagementService
+                .GetUser(id)
+                .HandleFailuresOrOk();
+        }
+
+        [HttpPut("users/{userId}")]
+        public async Task<ActionResult<Unit>> UpdateUser(string userId, UserEditViewModel model)
+        {
+            return await _userManagementService
+                .UpdateUser(userId, model.RoleId)
+                .HandleFailuresOrOk();
+        }
+
+        [HttpPost("users/{userId:guid}/publication-role")]
+        [ProducesResponseType(200)]
+        public async Task<ActionResult<Unit>> AddPublicationRole(Guid userId, AddPublicationRoleViewModel request)
+        {
+            return await _userRoleService
+                .AddPublicationRole(userId, request.PublicationId, request.PublicationRole)
+                .HandleFailuresOrOk();
+        }
+
+        [HttpPost("users/{userId:guid}/release-role")]
+        [ProducesResponseType(200)]
+        public async Task<ActionResult<Unit>> AddReleaseRole(Guid userId, AddReleaseRoleViewModel request)
+        {
+            return await _userRoleService
+                .AddReleaseRole(userId, request.ReleaseId, request.ReleaseRole)
+                .HandleFailuresOrOk();
+        }
+
+        [HttpDelete("users/publication-role/{id:guid}")]
+        [ProducesResponseType(200)]
```

Comment:
I'm not sure whether to change this or not because I think it returns a JSON object `{}`. I'm tempted to strip out these because we don't have them in many other places?